src/app/components/bookmark-menu/bookmark-menu.component.ts

```diff
@@ -36,31 +38,46 @@ export class BookmarkMenuComponent implements OnInit {
 
   public topBookmarks: Array<BookMark>;
   public isBookmarkOpen = false;
+  public currentCollections = '';
 
   constructor(
     public dialog: MatDialog,
-    private bookmarkService: ArlasBookmarkService
+    private bookmarkService: ArlasBookmarkService,
+    private startupService: ArlasStartupService
   ) { }
 
   public ngOnInit(): void {
     this.icon = this.icon ? this.icon : 'view_list';
     this.nbTopBookmarks = this.nbTopBookmarks ? this.nbTopBookmarks : 3;
+    this.currentCollections = Array.from(this.startupService.collectionsMap.keys()).join(',');
 
     if (this.bookmarkService.dataBase instanceof BookmarkPersistenceDatabase) {
       (this.bookmarkService.dataBase as BookmarkPersistenceDatabase).dataChange
         .subscribe((bookmarks: { total: number, items: BookMark[] }) => {
-          const sortedBookmark = bookmarks.items.sort((a, b) => {
+          const sortedBookmark = bookmarks.items.filter(bk => {
+            if (bk.collections) {
+              return bk.collections === this.currentCollections;
+            } else {
+              return false;
+            }
+          }).sort((a, b) => {
             return (a.views < b.views ? -1 : 1) * (-1);
```

Comment:
Usually sort can be done with `return  a.views - b.views`(or the contrary). Any the `*(-1)`looks like the check should be `a.views > b.views`